### DISCUSSION TITLE
Is it possible or normal to get <BS> (backspace) not to terminate?

### DISCUSSION THREAD

**Discussion by unphased** [dcampos/nvim-snippy#132] [Q&A] [answered]
I'm using snippy with cmp-snippy. 

I would like to prepopulate and give default values for my snippets because they are a nice hint. However even when I try stuff like this: 

```
  mapping = cmp.mapping.preset.insert({
...
    ["<BS>"] = cmp.mapping(function(fallback)
      if require'snippy'.is_active() then
        require'snippy'.cut_text('char', true)
      else
        fallback()
      end
    end, { "i", "s" }),
```
I cannot get clean behavior. Sometimes this works but sometimes it causes more actions to take place and still terminates the snippet expansion state.

I also tried

```
require('snippy').setup({
  mappings = {
    is = {
      ['<C-Tab>'] = 'expand_or_advance',
      ['<C-S-Tab>'] = 'previous',
      ['<BS>'] = 'cut_text'
    },
...
```
This does *something*, it prevents backspace from doing anything. It feels like the same behavior I had when I attempted to use the change up above and didnt send anything into cut_text().

I also made sure to remove any remaps for `<BS>` from other plugins. I'll first use a completely unused map though to test out the behavior before trying to bind it back to `<BS>`...

I am under the impression (from past usage of other snippet systems) that default values should be possible to clear out and I can continue to tab through the stops until I'm done editing.

For now the workaround involves two methods: 

- leave the deletion as the last thing I do, because it deactivates after i type `<BS>`.
- when in selection mode on a prepopulated item, type a single letter and `<BS>` to clear it back out, this still lets me continue.

**Comment by unphased** [comment]
I also tried 

```
  mapping = cmp.mapping.preset.insert({
...
    ["<BS>"] = cmp.mapping(function(fallback)
      if require'snippy'.is_active() then
        vim.api.nvim_feedkeys('a', 's', true)
        fallback()
      else
        fallback()
      end
    end, { "s" }),
```

In hopes of replicating what would happen if i type a letter followed by `<BS>`.

This is a bit closer but also seems to sometimes terminate the snippet expansion context.

**Comment by dcampos** [comment, marked as answer]
I have these select mode mappings in my config for so long that I don't remember where I took them from, but all they do is make `<BS>`, `<c-h>` and `<Delete>` work properly in select mode:

```lua
vim.keymap.set('s', '<BS>', '<c-g>"_c')
vim.keymap.set('s', '<c-h>', '<c-g>"_c')
vim.keymap.set('s', '<Delete>', '<c-g>"_c')
```

I hope they solve it for you too.

**Reply by unphased** [reply]
Ok I need to try these out!! Thank you!
